src/index.js

```diff
@@ -0,0 +1,181 @@
+
+  updateTempColor(newTemp);
+}
+
+const updateTempColor = (temperature) => {
+  tempValue.classList.remove('red','orange','yellow','green','turqoise');
+
+  if (temperature > 79) {
+    tempValue.classList.add('red');
+  } else if (temperature > 69) {
+    tempValue.classList.add('orange');
+  } else if (temperature > 59) {
+    tempValue.classList.add('yellow');
+  } else if (temperature > 49) {
+    tempValue.classList.add('green');
+  } else {
+    tempValue.classList.add('turqoise');
+  }
+
+  updateLandscape();
+}
+
+const updateLandscape = () => {
+  let currentTemp = parseInt(tempValue.textContent);
+
+  if (currentTemp >= 79) {
+    landscape.textContent = '🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂';
+  } else if (currentTemp >= 69) {
+    landscape.textContent = '🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷';
+  } else if (currentTemp >= 59) {
+    landscape.textContent = '🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃';
+  } else {
+    landscape.textContent = '🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲';
+  }
+}
+
+// WAVE 3
+const updateCityName = () => {
+  cityNameInput.addEventListener("input", updateValue);
+
+  function updateValue(e) {
+    headerCityName.textContent = e.target.value;
+  }
+}
+
+// WAVE 4
+const findLatitudeAndLongitude = async (query) => {
+  try {
+    const response = await axios
+      .get('http://127.0.0.1:5000/location', {
+        params: {
+          q: query,
+          format: 'json'
+        }
+      });
+
+    const latitude = response.data[0].lat;
+    const longitude = response.data[0].lon;
+    console.log('success in findLatitudeAndLongitude!', 'latitude:', latitude, 'longitude:', longitude);
+
+    return await findWeather(latitude, longitude);
+  } catch (error) {
+    console.log(error);
+    return null;
+  }
+}
+
+const findWeather = async (lat, lon) => {
+  try {
+    const response = await axios
+      .get('http://127.0.0.1:5000/weather', {
+        params: {
+          format: 'json',
+          lat: lat,
+          lon: lon,
+        }
+      });
+
+    console.log('success in findWeather!', response.data);
+    return response.data.main.temp;
+  } catch (error) {
+    console.log('error in findWeather!');
+    return null;
+  }
+}
+
+// WAVE 5
+const changeSky = () => {
+  skySelect.addEventListener('change', () => {
+    gardenSection.classList.remove('sunny','cloudy','rainy','snowy');
+
+    if (skySelect.value == 'default') {
+      sky.textContent = '';
+    } else if (skySelect.value == 'sunny') {
+      sky.textContent = `☁️ ☁️ ☁️ ☀️ ☁️ ☁️`;
+      gardenSection.classList.add('sunny');
+    } else if (skySelect.value == 'cloudy') {
+      sky.textContent =`☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️`;
+      gardenSection.classList.add('cloudy');
+    } else if (skySelect.value == 'rainy') {
+      sky.textContent = `🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧`;
+      gardenSection.classList.add('rainy');
+    } else if (skySelect.value == 'snowy') {
+      sky.textContent = `🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨`;
+        gardenSection.classList.add('snowy');
+    }
+  });
+}
+
+// WAVE 6
+const resetCityName = () => {
```

Comment:
Really nice use of helper functions to divide up the actions we want to take into smaller portions that are easier to read and test.